### PR TITLE
Implement dragging and dropping from OS file explorer

### LINKED ISSUE #570: Dragging files from OS into tree view yields error
I am using Xubuntu 14.04. When I drag a file from my desktop and drop it into a project (sub)folder in the atom treeview, I get the following two error messages ( /home/... is the project (sub)folder which I am dragging to ).

screen

### PR DESCRIPTION
**Pull request by Tyriar**
Fixes #570

### CODE REVIEW

**Comment by Tyriar** [comment]
Under Ubuntu 15.04

recording

**Comment by chuckhendo** [comment]
I was getting the same error on OS X El Capitan, and this PR fixes it

**Comment by Zireael07** [comment]
Has this been looked at at all?

**Comment by joshaber** [comment]
@Zireael07 Thanks for the nudge! It hasn't been reviewed yet but I'll try to get to it this week.

File: lib/tree-view.coffee [hunk 1]
```diff
@@ -832,11 +832,15 @@ class TreeView extends View
     entry = e.currentTarget
     return unless entry instanceof DirectoryView
 
-    initialPath = e.originalEvent.dataTransfer.getData("initialPath")
```
Comment (joshaber) [inline lib/tree-view.coffee]: It seems preferable to keep this, rather than looking it up twice below.

**Comment by joshaber** [comment]
Just one comment. Otherwise this looks good :+1:

**Comment by Tyriar** [comment]
@joshaber Good to go pending CI

**Comment by joshaber** [comment]
Seems to work as advertised. Thanks @Tyriar!

**Comment by Stefanbracke** [comment]
macos 10.12: Error while dropping file from an external volume.

EXDEV: cross-device link not permitted, rename '/Volumes/SomeFatHarddisk/spot1/banner_large.png' -> '/Users/dude/Documents/myproject/www/img/banner_large.png'

**Pull request merged**